From 2d83dff809f351f50ce6b6fcfadf303ff4520b59 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 12 Feb 2024 11:14:36 +0200
Subject: [PATCH] [SW-137121] gaudi3: erom pcie: Initialized efuse

efuse initialization required to check whether security enabled
as part of ppboot_security_init function.

Change-Id: I81aa98f33a37b0d9e51d75a71bb74b8bdd60c6a8
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index db7236a..8c060c3 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -156,6 +156,7 @@
 
 __weak void ppboot_security_init(void) {}
 __weak void ppboot_secure_arc(void) {}
+__weak void erom_efuse_init(void) {}
 
 void erom_main_function(void)
 {
@@ -163,6 +164,7 @@
 	int ret;
 
 	device_init();
+	erom_efuse_init();
 
 	/* Set ARCs to privileged mode */
 	ppboot_secure_arc();
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index dd0d645..e2dbccb 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1440,3 +1440,8 @@
 
 	return (config_efuse & EFUSE_SYSCFG_DISABLE_SECONDARY_IMG);
 }
+
+void erom_efuse_init(void)
+{
+	device_init_efuse();
+}
diff --git a/include/baremetal.h b/include/baremetal.h
index b33158e..3b57c83 100644
--- a/include/baremetal.h
+++ b/include/baremetal.h
@@ -284,6 +284,7 @@
 void preboot_recovery_flag(bool en);
 
 bool hl_is_efuse_syscfg_dis_secondary_img_set(void);
+void erom_efuse_init(void);
 
 #endif /* __ASSEMBLY__ */
 
